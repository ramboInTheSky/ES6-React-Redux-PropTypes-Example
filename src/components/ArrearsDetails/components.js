import styled from 'styled-components';
import { Button, Typography, mediaQueries } from 'nhh-styles';

export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;
