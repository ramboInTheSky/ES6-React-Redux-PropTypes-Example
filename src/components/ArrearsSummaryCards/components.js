import styled from 'styled-components';
import { mediaQueries } from 'nhh-styles';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 12px;

  > a {
    color: inherit;
    margin-bottom: 16px;
    margin-right: 1%;
    width: 32%;

    &:hover {
      color: inherit;
      text-decoration: none;
    }

    ${mediaQueries.mobile`
      margin-right: 0;
      width: 100%;
    `};

    ${mediaQueries.tablet`
      width: 49%;
    `};
  }
`;
