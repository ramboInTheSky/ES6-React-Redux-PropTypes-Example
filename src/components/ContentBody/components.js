import styled from 'styled-components';

export const OuterContainer = styled.div`
  padding-top: 50px;
  padding-bottom: 50px;
`;
export const InnerContainer = styled.div``;
