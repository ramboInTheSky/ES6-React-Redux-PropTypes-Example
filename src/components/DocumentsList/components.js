import styled from 'styled-components';
import { Button, mediaQueries, Typography } from 'nhh-styles';

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;

export const Link = styled(Button)`
  font-size: inherit;
  font-weight: 400;
  display: inline;
`;

export const Container = styled.section`
  padding: 25px 28px 0;

  ${mediaQueries.mobile`
    padding: 15px 18px;
  `};
`;

export const ErrorMessage = styled.div`
  color: ${({ theme }) => (theme ? theme.colors.support.two : 'inherit')};
  margin-bottom: 50px;
`;

export const DateField = styled.div``;

export const Row = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.colors.tertiaryLighter};
  color: ${({ theme }) => theme.colors.tertiary};
  display: flex;
  flex-direction: row;
  font-size: 16px;
  padding: 8px 0;
`;

export const Header = Row.extend`
  font-size: 14px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const Column = styled.div`
  width: ${({ width }) => width || 'auto'};
  text-align: ${({ align }) => align || 'left'};
  padding-right: 8px;
  word-wrap: break-word;

  &:last-child {
    padding-right: 0;
  }

  ${mediaQueries.mobile`
    padding-right: 4px;
  `};
`;

export const DesktopView = styled.div`
  display: none;

  ${mediaQueries.desktop`
    display: block;
  `};
`;

export const MobileAndTabletView = styled.div`
  display: block;

  ${mediaQueries.desktop`
    display: none;
  `};
`;
