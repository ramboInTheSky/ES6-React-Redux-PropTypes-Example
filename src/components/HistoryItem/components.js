import styled from 'styled-components';
import { Typography } from 'nhh-styles';

export const Label = styled.span`
  font-weight: 700;
`;

export const Title = styled(Typography.H3)`
  margin: -30px 0 30px 0;
`;

export const Value = styled.span`
  word-break: break-word;
`;
