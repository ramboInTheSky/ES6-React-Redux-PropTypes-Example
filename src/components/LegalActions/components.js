import styled, { withTheme } from 'styled-components';
import { Typography } from 'nhh-styles';

export const Wrapper = styled.div`
  margin: 1.5rem 0 1.875rem 0;
`;

export const P = styled(Typography.P)`
  margin: 1rem 0;
`;

export const Label = styled.span`
  font-weight: 700;
`;

export const Highlight = withTheme(styled.span`
  color: ${({ theme, isExpiring }) => (isExpiring ? theme.colors.support.two : 'inherit')};
  ${({ isExpiring }) => isExpiring && `font-weight: 700;`};
`);
