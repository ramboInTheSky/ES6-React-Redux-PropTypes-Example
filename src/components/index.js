export { default as ArrearsDetails } from './ArrearsDetails';
export { default as ArrearsSummaryCards } from './ArrearsSummaryCards';
export { default as ButtonRow } from './ButtonRow';
export { default as ContentBody } from './ContentBody';
export { default as ErrorBoundary } from './ErrorBoundary';
export { default as FieldRow } from './FieldRow';
export { default as FormWrapper } from './FormWrapper';
export { default as Heading } from './Heading';
export { default as HistoryItem } from './HistoryItem';
export { default as LegalActions } from './LegalActions';
export { default as MediaTableView } from './MediaTableView';
export { default as ModalFieldRow } from './ModalFieldRow';
export { default as NoTouching } from './NoTouching';
export { default as PreviewLinkContainer } from './PreviewLinkContainer';
export { default as PreviewTemplate } from './PreviewTemplate';
export { default as PreviewLink } from './PreviewLink';
export { default as NotificationWrapper } from './NotificationWrapper';
export { default as TaskTitle } from './TaskTitle';
export { default as TasksSummaryCards } from './TasksSummaryCards';
export { default as TemplateContainer } from './TemplateContainer';
