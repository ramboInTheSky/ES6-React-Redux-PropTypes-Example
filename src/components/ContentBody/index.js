import React from 'react';
import PropTypes from 'prop-types';

import { InnerContainer, OuterContainer } from './components';

const ContentBody = ({ children }) => (
  <OuterContainer className="container-fluid">
    <InnerContainer className="container">{children}</InnerContainer>
  </OuterContainer>
);

ContentBody.defaultProps = {
  children: null,
};

ContentBody.propTypes = {
  children: PropTypes.node,
};

export default ContentBody;
