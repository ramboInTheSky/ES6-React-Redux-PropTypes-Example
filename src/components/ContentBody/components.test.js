import React from 'react';
import { shallow } from 'enzyme';

import 'jest-styled-components';

import { InnerContainer, OuterContainer } from './components';

describe('Page components', () => {
  it('should render <InnerContainer /> correctly', () => {
    expect(shallow(<InnerContainer />)).toMatchSnapshot();
  });
  it('should render <OuterContainer /> correctly', () => {
    expect(shallow(<OuterContainer />)).toMatchSnapshot();
  });
});
