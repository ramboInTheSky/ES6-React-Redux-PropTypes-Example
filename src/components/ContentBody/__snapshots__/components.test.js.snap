// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page components should render <InnerContainer /> correctly 1`] = `
<div
  className=""
/>
`;

exports[`Page components should render <OuterContainer /> correctly 1`] = `
.c0 {
  padding-top: 50px;
  padding-bottom: 50px;
}

<div
  className="c0"
/>
`;
